components/cam/src/physics/crm/sam/SGS_TKE/diffuse_mom2D.F90

```diff
@@ -62,22 +74,47 @@ subroutine diffuse_mom2D(ncrms,grdf_x, grdf_z, dimx1_d, dimx2_d, dimy1_d, dimy2_
     end do
     !For working around PGI bugs where PGI did not allocate enough gangs
     numgangs = ceiling( ncrms*nzm*nx/128. )
+#if defined(_OPENACC)
     !$acc parallel loop gang vector collapse(3) vector_length(128) num_gangs(numgangs) async(asyncid)
+#elif defined(_OPENMP)
+    !$omp target teams distribute parallel do collapse(3)
+#endif
     do k=1,nzm
       do i=1,nx
         do icrm = 1 , ncrms
           kc=k+1
           ib=i-1
-          dudt(icrm,i,j,k,na)=dudt(icrm,i,j,k,na)-(fu(icrm,i,j,k)-fu(icrm,ib,j,k))
-          dvdt(icrm,i,j,k,na)=dvdt(icrm,i,j,k,na)-(fv(icrm,i,j,k)-fv(icrm,ib,j,k))
-          dwdt(icrm,i,j,kc,na)=dwdt(icrm,i,j,kc,na)-(fw(icrm,i,j,k)-fw(icrm,ib,j,k))
+          dfu = (fu(icrm,i,j,k)-fu(icrm,ib,j,k))
+          dfv = (fv(icrm,i,j,k)-fv(icrm,ib,j,k))
```

Comment:
This doesn't appear to be a case that needs atomics. You're allowed to read from i and i-1 in a loop so long as you aren't writing to the same variable. fu, fv, and fw are read-only, so it's OK to read from i and i-1.